### ISSUE TITLE
HSE calculation is interrupted without error reports

### ISSUE THREAD

**Issue by pxlxingliang** [deepmodeling/abacus-develop#4310]
### Describe the Testing Issue

An issue from [MumuMaster2021](https://github.com/MumuMaster2021)
 https://github.com/abacusmodeling/abacus-develop/issues/323

The error message is:
```
Abort(403294210) on node 0 (rank 0 in comm 0): Fatal error in internal_Isend: Unknown error class, error stack:
internal_Isend(14967): MPI_Isend(buf=0x14f7aec65010, count=-1724273464, MPI_CHAR, 0, 0, MPI_COMM_WORLD, request=0x18302da00) failed
internal_Isend(14911): Negative count, value is -1724273464
```

### Task list for Issue attackers (only for developers)

- [ ] Understand the testing issue described by the developer.
- [ ] Review the specific test case, expected and actual results, and any error messages.
- [ ] Identify the root cause of the test failure or issue.
- [ ] If a possible solution is suggested, evaluate its feasibility and effectiveness.
- [ ] Implement a fix for the test failure or issue, or create a new test case if needed.
- [ ] Verify that the fix resolves the testing issue and the test case passes.
- [ ] Review and update any relevant documentation, such as test plans or user guides.
- [ ] Ensure the testing issue is resolved and close the ticket.
- [ ] Share any lessons learned or best practices with the team to prevent similar issues in the future.

**Comment by pxlxingliang** [comment]
The same bug is in issue https://github.com/deepmodeling/abacus-develop/issues/3674.

**Comment by PeizeLin** [comment]
The bug has been fixed in the 2023.11.06 version of LibComm.
